src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -296,6 +296,12 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
                 return false;
             }
 
+            if (analyzer is ArtifactGenerator)
```

Comment:
The changes to `CompilationWithAnalyzers.cs` to always pass in null for `artifactCallback` indicate this is a build-only feature, is that correct? If so, we should make it explicit in the design proposal.

Given that is intentional, I think we should also check `artifactCallback` is non-null in this condition. This will ensure that when running analyzers from the IDE (via CompilationWithAnalyzers), we only run artifact generators if they are reporting diagnostics. Otherwise, we will end up with redundant callbacks into artifact generators.

Reply:
> If so, we should make it explicit in the design proposal.

Sure.